dataflowAPI/src/RoseImpl.C

```diff
@@ -838,7 +838,7 @@ SgAsmBinaryMultiply::SgAsmBinaryMultiply(SgAsmExpression *lhs, SgAsmExpression *
 }
 
 SgAsmType *SgAsmBinaryMultiply::get_type() const {
-    return ((SgAsmBinaryExpression *) this)->get_type();
+    return static_cast<const SgAsmBinaryExpression *>(this)->get_type();
```

Comment:
I'm not disagreeing with the change here, but should we actually fix this and use the type-safe `return SgAsmBinaryExpression::get_type();`?

Reply:
The code as written results in infinite recursion

Reply:
I will rewrite this and SgAsmBinaryAdd::get_type to call `SgAsmBinaryExpression::get_type()`